src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,343 @@
+            AbstractFeatureReader<VariantContext, LineIterator> reader = AbstractFeatureReader.getFeatureReader(variantURI, null, new VCFCodec(), true, cloudWrapper, cloudIndexWrapper);
+
+            // A GVCF file must contain only one sample, throw an exception otherwise
+            if (((VCFHeader)reader.getHeader()).getGenotypeSamples().size() > 1) {
+                throw new UserException("Input GVCF: " + variantPath + " contains more than one sample");
+            }
+
+            String sampleName = ((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0);
+            sampleToVCFMap.put(sampleName, reader);
+        }
+
+        createMergedHeader();
+
+        initializeIntervals();
+
+        super.onStartup();
+    }
+
+    /**
+     * Create merged header from input GVCFs
+     */
+    private void createMergedHeader() {
+        List<VCFHeader> headers = new ArrayList<>();
+        for (Map.Entry<String, FeatureReader<VariantContext>> sampleToVCF : sampleToVCFMap.entrySet()) {
+            FeatureReader<VariantContext> reader = sampleToVCF.getValue();
+            VCFHeader vcfHeader = (VCFHeader)reader.getHeader();
+            headers.add((VCFHeader)reader.getHeader());
+        }
+
+        mergedHeader = VCFUtils.smartMergeHeaders(headers, true);
+        mergedVCFHeader = new VCFHeader(mergedHeader);
+    }
+
+    /**
+     * Loads our intervals using the best available sequence
+     * dictionary (as returned by {@link #getBestAvailableSequenceDictionary})
+     * to parse/verify them. Does nothing if no intervals were specified.
+     */
+    private void initializeIntervals() {
+        if (intervalArgumentCollection.intervalsSpecified()) {
+
+            final SAMSequenceDictionary intervalDictionary = getBestAvailableSequenceDictionary();
+            if (intervalDictionary == null) {
+                throw new UserException("We require at least one input source that " +
+                    "has a sequence dictionary (reference or reads) when intervals are specified");
+            }
+
+            intervals = new ArrayList<>();
+
+            List<SimpleInterval> simpleIntervalList =
+                intervalArgumentCollection.getIntervals(intervalDictionary);
+
+            if (simpleIntervalList.size() > 1) {
+                throw new UserException("More than one interval specified. The tool takes only one");
+            }
+
+            for (SimpleInterval simpleInterval : simpleIntervalList) {
+                intervals.add(new ChromosomeInterval(simpleInterval.getContig(),
+                  simpleInterval.getStart(), simpleInterval.getEnd()));
+            }
+        }
+    }
+
+    /**
+     * Since this tool merges headers from all input GVCFs, calling GATKTool's
+     * best available sequence directory is not efficient. If there are 50K samples,
+     * the existing method will look into each one of them to find the best available
```

Comment:
Method comment is not really accurate -- change to something like "Overriding getBestAvailableSequenceDictionary() to prefer the mergedVCFHeader's sequence dictionary, if present, over any other dictionaries"